### DISCUSSION TITLE
After upgrading from 0.11 to 0.12 I get "AttributeError: module 'typer' has no attribute 'Typer'"

### DISCUSSION THREAD

**Discussion by corneliusroemer** [fastapi/typer#799] [Questions]
Broken out from https://github.com/tiangolo/typer/discussions/784#discussioncomment-9106014

### Example Code

```python
import typer
app = typer.Typer(context_settings={"help_option_names": ["-h", "--help"]})
```

### Description

Getting error with 0.12.3, not with 0.11.1

```
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/bin/nwkfmt", line 3, in <module>
    from nwkfmt.cli import entry_point
  File "/Users/corneliusromer/code/nwk-formatter/nwkfmt/cli.py", line 7, in <module>
    app = typer.Typer(context_settings={"help_option_names": ["-h", "--help"]})
          ^^^^^^^^^^^
AttributeError: module 'typer' has no attribute 'Typer'
```

### Operating System

macOS

### Typer Version

0.12.3

### Python Version

3.11.8

**Comment by svlandeg** [comment]
Hi!

I can't reproduce this. Could you create an entirely fresh environment, `pip install typer`, run your example code, and show all the console output from these steps? Could you also share the output of `pip list`?

**Reply by corneliusroemer** [reply]
Not on desktop so can't share pip list right now. I installed Typer via conda-forge, maybe there is something wrong there with the packaging.

**Comment by samuller** [comment]
I use `poetry` and had the same error (only locally, not in CI). Tried `poetry install` a few times but didn't make a difference. Then I did `pip uninstall typer-slim` followed by `poetry install` and it worked again.

While investigating the issue I looked into the installed `typer` package's folder and it only contained the following files: `__init__.py`, `__main__.py`, `_typing.py`, `cli.py`, `rich_utils.py` and no other files (this is from memory, so I might've missed something). After I got it working I saw there were now many more files in that folder.

Here's also a `pip list` while it was failing in case that makes a difference:
```
$ pip list
Package           Version   Editable project location
----------------- --------- -------------------------
attrs             22.1.0
black             23.3.0
click             8.1.3
coverage          6.5.0
decorator         5.1.1
fastjsonschema    2.19.1
flake8            5.0.4
flake8-docstrings 1.6.0
greenlet          3.0.1
iniconfig         1.1.1
mccabe            0.7.0
mypy              1.4.1
mypy-extensions   1.0.0
networkx          2.5
packaging         24.0
pathspec          0.11.2
pgmerge           1.12.0    /workspace
pip               23.1
platformdirs      4.0.0
pluggy            1.0.0
psycopg2-binary   2.9.5
py                1.11.0
pycodestyle       2.9.1
pydocstyle        6.1.1
pyflakes          2.5.0
pyparsing         3.0.9
pytest            6.2.5
pytest-cov        3.0.0
PyYAML            6.0.1
rxjson            0.3
setuptools        67.6.1
snowballstemmer   2.2.0
SQLAlchemy        2.0.30
toml              0.10.2
tomli             2.0.1
typer-slim        0.12.3
types-PyYAML      6.0.12.12
typing_extensions 4.7.1
wheel             0.40.0
```

**Reply by svlandeg** [reply]
Thanks for the report @samuller! For us to be able to reproduce this, can you give the exact command (and their output) you ran on the console, starting from a fresh env? That'd be really helpful 🙏

**Comment by nannerpusser** [comment]
Just had this come up myself. I don't know where the responsibility lay (and sadly I lost the Terminal window that had every Traceback and command) but this came up for me after:

In a pretty large and well-used venv I had to reinstall `textual-pyfiglet` which lead to a full upgrade of `textual` and of course pulled in `rich` `rich-pyfiglet` and all the rest of the dependencies, including `typer`. Afterward, using pipenv and trying to sync and check everything in project, I was lead to safetycli.com and getsafety.com and during that auth and sign up, I ran `safety.exe auth <whatever>` locally and was thrown the error:

`AttributeError: module 'typer' has no attribute 'rich_utils'`

Which is why I mentioned all my rich/textual related stuff above. I wish I had the original Traceback for you, but that was the error verbatim and it lead me here. This was also from:

`Python312\site-packages\safety\cli.py`

Their CLI tool made with typer. In my case I got lucky and the line that tried calling `typer.rich_utils` was trying to <s>add a blank line to the CLI</s> print for pure cosmetic reasons and so I just commented the whole line out and it ran fine. If anyone ends up here in the future, it's line 1250 in the safety `cli.py` module in version `3.6.0`

other relevant versions:

pip 25.2 on python 3.12.8
typer 0.17.3
rich 14.1.0
Microsoft Windows 11 Pro Build 22631


I was able to fix this like I mentioned, by just removing the offending line from the Typer-based tool `safety` in my non-global AppData Python installation. 


Update, here is the offending portion of code that `safety` had used:

`cli_app = typer.Typer(rich_markup_mode="rich", cls=SafetyCLISubGroup)`
`#typer.rich_utils.STYLE_HELPTEXT = ""`

Lines 1249 and 1250 in their cli.py module.
